### PR TITLE
Site: Fixes

### PR DESCRIPTION
**Pull request by guykhazma**
This PR includes some minor fixes to the documentation site:

- Javadoc bug which makes the search form unusable - 
When using JDK 11 the Javadoc is generated using HTML 5 and has a search bar, due to a bug some of the links redirect to invalid locations, for example:
```
https://iceberg.apache.org/javadoc/0.9.0/undefined/org/apache/iceberg/Transaction.html
```
instead of:
```
https://iceberg.apache.org/javadoc/0.9.0/org/apache/iceberg/Transaction.html
```
The bug was fixed manually for version 0.9.0 and for future release I have modified the gradle task to make the necessary changes according to the solution suggested [here](https://stackoverflow.com/questions/52326318/maven-javadoc-search-redirects-to-undefined-url/52603413#52603413).

- Java API page changes:
  - Adding documentation for transactions support
  - Adding documentation for `iceberg-spark3`

- Minor typo fixes

### CODE REVIEW

File: tasks.gradle [hunk 1]
```diff
@@ -25,6 +25,20 @@ task aggregateJavadoc(type: Javadoc) {
...
   classpath = project.rootProject.files(subprojects.javadoc.classpath)
+
+  final JAVADOC_FIX_SEARCH_STR = '\n\n' +
+          'getURLPrefix = function(ui) {\n' +
+          '    return \'\';\n' +
+          '};\n'
+
+  doLast {
+    // Fix bug with search
+    if (JavaVersion.current() >= JavaVersion.VERSION_11) {
```
Comment (rdblue) [inline tasks.gradle:36]: Maybe we should just require Javadoc to be built with Java 8 instead?

File: site/docs/api.md [hunk 2]
```diff
@@ -99,6 +99,8 @@ Available operations to update a table are:
 
 ### Transactions
 
+`Table` also enables to commit multiple table operations at once.
```
Comment (rdblue) [inline site/docs/api.md:102]: I think we should either remove this empty section or expand it a bit beyond what you have here.

Transactions are used to commit multiple table changes in a single atomic operation. A transaction is used to create individual operations using factory methods, like `newAppend`, just like working with a `Table`. But operations created by a transaction are committed as a group when `commitTransaction` is called.

```java
Transaction t = table.newTrasaction();

// commit operations to the transaction
t.newDelete().deleteFromRowFilter(filter).commit();
t.newAppend().appendFile(data).commit();

// commit all the changes to the table
t.commitTransaction();
```

File: tasks.gradle (hunk 1 above)
Comment (guykhazma) [reply tasks.gradle:36]: The search is actually very nice addition, we can remove this fix and fix each time manually

Comment (rdblue) [reply tasks.gradle:36]: Oh, the search is only available with Java 11? Then that's a nice reason to keep it as you have it here.

File: site/docs/api.md (hunk 2 above)
Comment (guykhazma) [reply site/docs/api.md]: thanks for the suggestion. updated accordingly.

File: tasks.gradle (hunk 1 above)
Comment (guykhazma) [reply tasks.gradle:36]: the search was added starting with Java 9, it became default in Java 10 (see [here](http://openjdk.java.net/jeps/224))

**Comment by rdblue** [comment]
Looks great, I'll merge. Thanks for the contribution, @guykhazma!

**Pull request merged**